### PR TITLE
Adding info to debug WindowsAlternateDataStreamOverwrite failure

### PR DESCRIPTION
**Pull request by PranavSenthilnathan**
Adding info for: #83659 

- Adding assertion on HRESULT in the non-overwrite case to confirm that it does not have the same issue as the WindowsAlternateDataStreamOverwrite
- Adding logging of attributes
- Assertion on contents of files to determine whether everything is written/accessible before copy

I wasn't able to repro this after trying to run nativeaot build on arm64. Confirmed that it wasn't FAT32 issue (tried mounting volume and it gives a different error) and that the tests were running nativeaot (checked PlatformDetection.IsNativeAot). Hopefully adding the above will give some more info next time this fails.

### CODE REVIEW

File: src/libraries/System.Runtime/tests/System.IO.FileSystem.Tests/File/Copy.cs [hunk 1]
```diff
@@ -352,6 +363,8 @@ public void WindowsAlternateDataStreamOverwrite(string defaultStream, string alt
 
             // Copy the default stream into an alternate stream
             File.WriteAllText(testFileDefaultStream, "Foo");
+            Assert.Equal("Foo", File.ReadAllText(testFileDefaultStream));
+            Console.WriteLine($"Copying default stream into alternate stream.");
```
Comment (adamsitnik) [inline src/libraries/System.Runtime/tests/System.IO.FileSystem.Tests/File/Copy.cs:367]: I am not sure if the output written to the console will be accessible in the logs. IIRC the xUnit recommended approach is to add a public ctor that accepts `ITestOutputHelper`, store it in the field and then use it for writing in tests.

Sth like here:

https://github.com/dotnet/runtime/blob/d2a2a79b2eaf66d68a02f35917907bd9fbe97ea7/src/libraries/System.Private.Xml/tests/Xslt/XslTransformApi/CThreads.cs#L18-L21

**Review by adamsitnik — COMMENTED** [review]
It's good idea to log more info, I am just not sure if the current implementation is going to work.

File: src/libraries/System.Runtime/tests/System.IO.FileSystem.Tests/File/Copy.cs (hunk 1 above)
Comment (PranavSenthilnathan) [reply src/libraries/System.Runtime/tests/System.IO.FileSystem.Tests/File/Copy.cs]: Changed to use `ITestOutputHelper` and it works.

**Review by adamsitnik — APPROVED** [review]
LGTM, thank you @PranavSenthilnathan ! If possible please merge it at the beginning of your work day and take a look at the CI in a few hours, just to ensure the changes have not made the CI flaky.

**Pull request merged**